src/main/java/org/spongepowered/common/service/scheduler/AsyncScheduler.java

```diff
@@ -194,520 +113,73 @@ private void recalibrateMinimumTimeout() {
-     *     PluginContainer pluginContainer = result.get();
-     * </code>
-     * </pre>
-     *
-     * </p>
-     *
-     * <p>
-     * Example for specifying a certain time unit scale:
-     * </p>
-     *
-     * <p>
-     *
-     * <pre>
-     *     <code>
-     *         // The task will run with a period of 15 seconds.
-     *         runRepeatingTask(somePlugin, someRunnableTarget, TimeUnit.SECONDS, 15);
-     *     </code>
-     * </pre>
-     *
-     * </p>
-     *
-     * <p>
-     * Example for specifying a certain time unit scale:
-     * </p>
-     *
-     * <p>
-     *
-     * <pre>
-     *     <code>
-     *         // The task will run with a period of 120 milliseconds and delay of 302 milliseconds
-     *         // (If the scales are the same for both arguments)
-     *         runRepeatingTaskAfter(somePlugin, someRunnableTarget, TimeUnit.MILLISECONDS, 120, 302);
-     *
-     *         // If the two units are in different scales:
-     *
-     *         // The task will run with a period of 20 seconds and delay of 500 milliseconds:
-     *         Either:
-     *         runRepeatingTaskAfter(somePlugin, someRunnableTarget, TimeUnit.MILLISECONDS, TimeUnit.SECOND.toMillis(20), 500);
-     *         // OR
-     *         runRepeatingTaskAfter(somePlugin, someRunnableTarget, TimeUnit.SECONDS, 20, TimeUnit.MILLISECONDS.toSeconds(500));
-     *
-     *     </code>
-     * </pre>
-     *
-     * </p>
-     *
-     * @param plugin The plugin container of the Plugin that initiated the Task
-     * @param runnableTarget The Runnable object that implements a run() method
-     *        to execute the Task desired
-     * @param scale Time unit used to specify {@code interval} and {@code delay}
-     * @param delay The offset in time unit scale before running the task.
-     * @param interval The offset in time unit scale before running the task.
-     * @return Optional&lt;Task&gt; Either Optional.absent() if invalid or a
-     *         reference to the new Task
-     */
     @Override
-    public Optional<Task> runRepeatingTaskAfter(Object plugin, Runnable runnableTarget, TimeUnit scale, long interval, long delay) {
-        Optional<Task> resultTask = Optional.absent();
-
-        // The interval and delay passed to this method is converted to the number of milliseconds
-        // per the scale of the time unit.
-        interval = scale.toMillis(interval);
-        delay = scale.toMillis(delay);
-        ScheduledTask repeatingTask = this.schedulerHelper.taskValidationStep(plugin, runnableTarget, delay, interval);
-
-        if (repeatingTask == null) {
-            Sponge.getLogger().warn(SchedulerLogMessages.CANNOT_MAKE_TASK_WARNING);
-        } else {
-            resultTask = utilityForAddingAsyncTask(repeatingTask);
-        }
-
-        return resultTask;
+    protected void executeTaskRunnable(Runnable runnable) {
+        this.executor.submit(runnable);
```

Comment:
I have compensated for this by wrapping the runnable in another runnable, https://github.com/SpongePowered/SpongeCommon/pull/47/files#diff-168ace8114caeb27d02b38cad72bc18eR245